content/blog/options/index.md

```diff
@@ -0,0 +1,237 @@
+If we had an `Err` variant, we would just use `Result` in the first place.
+But there are cases where `None` is a completely valid value and we just don't have an error to return.
+We can just say "there is no value" and that's it.
+
+## What's The Solution?
+
+There are multiple solutions!
+
+The initial error message, while cryptic, gave us a hint:
+
+```rust
+use `.ok_or(...)?` to provide an error compatible with `Result<(), Box<dyn std::error::Error>>`.
+```
+
+Apparently we can use the `ok_or` method, which converts the `Option` into a `Result`:
+
+```rust
+let user = get_user().ok_or("No user")?;
+```
+
+Unfortunately, `ok_or` might not be the most discoverable name for this method.
+I find it unintuitive and needed to look it up many times.
+That's because `Ok` is commonly associated with the `Result` type, not `Option`.
+There's `Option::Some`, so it could also be called `some_or`, but that would be hard to change now.
+It was [actually suggested in 2014](https://github.com/rust-lang/rust/pull/17469#issuecomment-56919911), but the name `ok_or` won out,
+because `ok_or(MyError)` reads nicely and I can see that. Guess we have to live with the minor inconsistency now.
+
+Another downside is that `ok_or` *always* allocates a new `String` for the error message &ndash; even if we don't use it.
+If you want to avoid this allocation, you can use `ok_or_else`:
+
+```rust
+let user = get_user().ok_or_else(|| "No user".into())?;
+```
+
+This takes a closure that returns the error message.
+The closure only gets called if `get_user()` returns `None`.
+
+We avoid the allocation if not needed, but that's even more verbose!
+
+## Just Use `match`
+
+In the past, I used to recommend people to not be clever and just use a `match` statement.
+
+```rust
+let user = match get_user() {
+    Some(user) => user,
+    None => return Err("No user".into()),
+};
+```
+
+`match` works in combination with `Option`, because it's just an enum and we can pattern match on it.
+As long as we cover all cases, the compiler is happy. In this situation, we only have two cases: `Some` and `None`.
+In the `None` case, we return early with an error. In the `Some` case, we continue with the value.
+(`match` is an expression and the value of the last expression in the block is returned. In our case it's `user` and it
+gets assigned to the `user` variable in the outer scope.)
+
+This is more explicit and easier to understand for beginners.
+The one issue I had when teaching this was that it looked a bit more verbose for simple cases like this.
+
+## The Best Solution To Handle None With The Standard Library
+
+Recently, the `let-else` expression was stabilized, so now you can write this: 
+
+```rust
+let Some(user) = get_user() else {
+    return Err("No user".into());
+};
+
+// Do something with user
+```
+
+In my opinion, that's the best of both worlds: it's compact and easy to understand.
+It's unanimously loved by beginners and experienced Rustaceans alike.
```

Comment:
I would also mention it highlights the "happy path" by making it the obvious flow of the program. With match you don't know which arm of the statement is the intended one until you read both expressions. Here it's clear we want the value, and only do something _in case_ we don't have one. I think Golang had a mantra about that somewhere too.

Reply:
done! great idea!